### PR TITLE
Feature/chatgpt sample i18n

### PR DESCRIPTION
**Pull request by morehardy**
fix: text erorr of chatgpt-sample's wala game
chore: chatgpt sample json generator supports i18n
feat: chatgpt sample supports i18n

### CODE REVIEW

**Comment by phodal** [comment]
build failured

**Comment by morehardy** [comment]
> build failured

A magic bug. This build error causes from I add a line of 'import type...'. It works after I change the 'import type...' to 'import ...'. I'll resubmit a pull request.

File: src/app/[lang]/chatgpt-samples/[id]/page.tsx [hunk 1]
```diff
@@ -17,23 +17,12 @@ import { notFound } from "next/navigation";
...
-  category: string;
-  author: string;
-  homepage: string;
-  preview: string;
-  steps: {
-    ask: string;
-    response: string;
-  }[];
-}
+import type { Sample, SampleDetail } from "../type";
```
Comment (CGQAQ) [inline src/app/[lang]/chatgpt-samples/[id]/page.tsx:20]: your type Sample

Comment (CGQAQ) [inline src/app/[lang]/chatgpt-samples/[id]/page.tsx:20]: your type `Sample`

File: src/app/[lang]/chatgpt-samples/[id]/page.tsx [hunk 2]
```diff
@@ -17,23 +17,12 @@ import { notFound } from "next/navigation";
...
 
 const getSampleNames = async () => {
-  const index = await import("@/assets/chatgpt/samples/index.json").then((mod) => mod.default);
-  return index.map((item) => item.path.split(".").slice(0, -1).join("."));
+  const { locale } = await getAppData();
+  const index = await import(`@/assets/chatgpt/samples/index_${locale}.json`).then((mod) => mod.default);
+  return index.map((item: Sample) => item.path.split(".").slice(0, -1).join("."));
 };
 
 async function Sample({ params }: { params: { id: string } }) {
```
Comment (CGQAQ) [inline src/app/[lang]/chatgpt-samples/[id]/page.tsx:28]: and the component name

**Review by CGQAQ — COMMENTED** [review]
your type and component name is the Same, in the below, you have a 
```tsx
export default Sample;
```
this line makes typescript compiler confused

File: src/app/[lang]/chatgpt-samples/[id]/page.tsx (hunk 2 above)
Comment (morehardy) [reply src/app/[lang]/chatgpt-samples/[id]/page.tsx:28]: Got it. Thanks for your help.

**Pull request closed without merging**